backend/MonChenil/Services/TimeSlotService.cs

```diff
@@ -58,9 +64,67 @@ private void CheckTimeSlot(TimeSlot timeSlot)
         }
     }
 
+    public void BookTimeSlotForPets(TimeSlot timeSlot, IEnumerable<Pet> pets)
+    {
+        if (timeSlot == null)
+        {
+            throw new ArgumentNullException(nameof(timeSlot));
+        }
+
+        if (pets == null || !pets.Any())
+        {
+            throw new ArgumentException("At least one pet must be selected for booking.", nameof(pets));
+        }
+
+        foreach (var pet in pets)
+        {
+            foreach (var existingPet in timeSlot.Pets)
+            {
+                if (existingPet.IncompatibleTypes.Contains(pet.Type))
+                {
+                    throw new ArgumentException($"The pet '{pet.Name}' (Type: {pet.Type}) is not compatible with another pet already booked for this time slot.");
+                }
+            }
+        }
+
+        timeSlot.Pets.AddRange(pets);
+
+        _repository.Update(timeSlot);
+    }
+
+    public void AddPetToTimeSlot(TimeSlot timeSlot, IEnumerable<Pet> pets)
+    {
+        if (timeSlot == null)
+        {
+            throw new ArgumentNullException(nameof(timeSlot));
+        }
+
+        if (pets == null || !pets.Any())
+        {
+            throw new ArgumentException("At least one pet must be selected for booking.", nameof(pets));
+        }
+
+        foreach (var pet in pets)
+        {
+            if (timeSlot.Pets.Any(existingPet => ArePetsIncompatible(existingPet, pet)))
+            {
+                throw new ArgumentException("Incompatible pets cannot be added to the same time slot.", nameof(pets));
+            }
+        }
+
+        timeSlot.Pets.AddRange(pets);
+
+        _repository.Update(timeSlot);
+    }
+
+    private bool ArePetsIncompatible(Pet pet1, Pet pet2)
```

Comment:
Rename to ArePetsCompatible()
Change calls from ArePetsIncompatible() to !ArePetsCompatible()